src/hotspot/share/memory/metaspaceShared.cpp

```diff
@@ -201,149 +201,29 @@ static char* compute_shared_base(size_t cds_total) {
 
 #ifdef _LP64
-  // On 64-bit VM we reserve a 4G range and, if UseCompressedClassPointers=1,
-  //  will use that to house both the archives and the ccs. See below for
-  //  details.
   const uint64_t UnscaledClassSpaceMax = (uint64_t(max_juint) + 1);
-  const size_t cds_total = align_down(UnscaledClassSpaceMax, reserve_alignment);
+  cds_max = align_down(UnscaledClassSpaceMax, reserve_alignment);
 #else
   // We don't support archives larger than 256MB on 32-bit due to limited
   //  virtual address space.
-  size_t cds_total = align_down(256*M, reserve_alignment);
+  cds_max = align_down(256*M, reserve_alignment);
 #endif
 
-  char* shared_base = compute_shared_base(cds_total);
-  _requested_base_address = shared_base;
-
-  // Whether to use SharedBaseAddress as attach address.
-  bool use_requested_base = true;
-
-  if (shared_base == NULL) {
-    use_requested_base = false;
-  }
-
-  if (ArchiveRelocationMode == 1) {
-    log_info(cds)("ArchiveRelocationMode == 1: always allocate class space at an alternative address");
-    use_requested_base = false;
-  }
-
-  // First try to reserve the space at the specified SharedBaseAddress.
-  assert(!_shared_rs.is_reserved(), "must be");
-  if (use_requested_base) {
-    _shared_rs = ReservedSpace(cds_total, reserve_alignment,
-                               false /* large */, (char*)shared_base);
-    if (_shared_rs.is_reserved()) {
-      assert(_shared_rs.base() == shared_base, "should match");
-    } else {
-      log_info(cds)("dumptime space reservation: failed to map at "
-                    "SharedBaseAddress " PTR_FORMAT, p2i(shared_base));
-    }
-  }
-  if (!_shared_rs.is_reserved()) {
-    // Get a reserved space anywhere if attaching at the SharedBaseAddress
-    //  fails:
-    if (UseCompressedClassPointers) {
-      // If we need to reserve class space as well, let the platform handle
-      //  the reservation.
-      LP64_ONLY(_shared_rs =
-                Metaspace::reserve_address_space_for_compressed_classes(cds_total);)
-      NOT_LP64(ShouldNotReachHere();)
-    } else {
-      // anywhere is fine.
-      _shared_rs = ReservedSpace(cds_total, reserve_alignment,
-                                 false /* large */, (char*)NULL);
-    }
-  }
-
-  if (!_shared_rs.is_reserved()) {
-    vm_exit_during_initialization("Unable to reserve memory for shared space",
-                                  err_msg(SIZE_FORMAT " bytes.", cds_total));
-  }
-
-#ifdef _LP64
-
-  if (UseCompressedClassPointers) {
-
-    assert(CompressedKlassPointers::is_valid_base((address)_shared_rs.base()), "Sanity");
-
-    // On 64-bit VM, if UseCompressedClassPointers=1, the compressed class space
-    //  must be allocated near the cds such as that the compressed Klass pointer
-    //  encoding can be used to en/decode pointers from both cds and ccs. Since
-    //  Metaspace cannot do this (it knows nothing about cds), we do it for
-    //  Metaspace here and pass it the space to use for ccs.
```

Comment:
So this whole comment is wrong now and obsolete?

Reply:
Yes, most of the work done under the old function `MetaspaceShared::initialize_dumptime_shared_and_meta_spaces()` are now unnecessary. Specifically, it doesn't initialize the "shared space" or "meta space" anymore. That's why I renamed it to `MetaspaceShared::initialize_for_static_dump()`. 

Now it only does the following:
- reset `SharedBaseAddress` if too high
- reserve `_symbol_rs`

===============
BTW, I have added comments at the top of archiveBuilder.hpp to give an overview of how the memory is reserved and relocated during the dump process.

Reply:
Thanks for the clarification.